models/issues/comment.go

```diff
@@ -1122,36 +1113,65 @@ func UpdateComment(ctx context.Context, c *Comment, contentVersion int, doer *us
 }
 
 // DeleteComment deletes the comment
-func DeleteComment(ctx context.Context, comment *Comment) error {
-	e := db.GetEngine(ctx)
-	if _, err := e.ID(comment.ID).NoAutoCondition().Delete(comment); err != nil {
-		return err
+func DeleteComment(ctx context.Context, comment *Comment) (*Comment, error) {
+	if _, err := db.GetEngine(ctx).ID(comment.ID).NoAutoCondition().Delete(comment); err != nil {
+		return nil, err
 	}
 
 	if _, err := db.DeleteByBean(ctx, &ContentHistory{
 		CommentID: comment.ID,
 	}); err != nil {
-		return err
+		return nil, err
 	}
 
 	if comment.Type.CountedAsConversation() {
 		if err := UpdateIssueNumComments(ctx, comment.IssueID); err != nil {
-			return err
+			return nil, err
 		}
 	}
-	if _, err := e.Table("action").
+	if _, err := db.GetEngine(ctx).Table("action").
 		Where("comment_id = ?", comment.ID).
 		Update(map[string]any{
 			"is_deleted": true,
 		}); err != nil {
-		return err
+		return nil, err
+	}
+
+	var deletedReviewComment *Comment
+
+	// delete review & review comment if the code comment is the last comment of the review
```

Comment:
Why it can delete the review? What if the review is a "change request" or "approval"?

Reply:
moved to #35133